### PR TITLE
[BEAM-1746] Include LICENSE and NOTICE in python dist files

### PR DESCRIPTION
**Pull request by aaltay**
R: @davorbonaci 

Davor, is this the right way to approach the problem? (I would like to copy files before packaging and delete copied files post packaging.)

### CODE REVIEW

File: sdks/python/pom.xml [hunk 1]
```diff
@@ -56,6 +56,41 @@
...
+            </execution>
+          </executions>
+      </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-antrun-plugin</artifactId>
+          <executions>
+            <execution>
+              <id>copy-project-files-clean</id>
+              <phase>clean</phase>
```
Comment (davorbonaci) [inline sdks/python/pom.xml]: This is much better served by configuring the maven-clean-plugin directly, as opposed to creating a new execution to delete a file.

File: sdks/python/pom.xml [hunk 2]
```diff
@@ -56,6 +56,41 @@
...
+          <executions>
+            <execution>
+              <id>copy-project-files</id>
+              <phase>prepare-package</phase>
+              <goals>
+                <goal>run</goal>
+              </goals>
+              <configuration>
+                <tasks>
+                  <copy file="../../LICENSE" tofile="${basedir}/LICENSE"/>
```
Comment (davorbonaci) [inline sdks/python/pom.xml]: I wouldn't do this -- reaching outside the current context generally breaks things.

**Comment by aaltay** [comment]
@davor updated the PR, PTAL.

This changes moves the copying piece to the root pom.xml. At root level these two files needed for Python packaging (LICENSE and NOTICE) pushed down to the right folder.

One may ask, why are we not packaging python at root instead of pushing down these files. The reason is we need to execute `setup.py` for packaging. The resulting package could only contain files that exist in the same sub-directory tree as the location of `setup.py`. And building the python package at root would require copying the whole `sdks/python` path to root.

Another question would be, what happens if `mvn` is invoked directly as `sdks/python` directory. In this case, packaging will succeed however resulting artifacts would not have the necessary files. This should not be a problem since our release procedure requires running `mvn` at root for packaging all needed artifacts.

**Comment by davorbonaci** [comment]
LGTM

**Pull request closed without merging**